web-api/src/notifications/sendNotificationToUserTypes.ts

```diff
@@ -0,0 +1,213 @@
+import { RawPractitioner } from '@shared/business/entities/Practitioner';
```

Comment:
I wonder if we could have:

```
type BaseNotification = {
  action: string; // Discriminant for narrowing types
};
```

And then extend each type as needed to avoid the long union type at the end.

Reply:
E.g.

```
type NotificationWithProgress = BaseNotification & {
  filesCompleted: number;
  totalFiles: number;
};
```

Reply:
If action is not specific string value we can mismatch required fields for the action being used, for example I can use action1 extra props for action2


```typescript
{
action: 'action1',
action2Prop: true
}

{
action: 'action2',
action1Prop: 'string'
}
``` 

and no errors will be thrown because it does not care for specific action as long as it fits one of the extension bodies.